src/sunadaptcontroller/imexgus/sunadaptcontroller_imexgus.c

```diff
@@ -94,6 +107,92 @@ SUNAdaptController SUNAdaptController_ImExGus(SUNContext sunctx)
   return (C);
 }
 
+/* -----------------------------------------------------------------
+ * Function to control ImExGus parameters from the command line
+ */
+
+SUNErrCode SUNAdaptController_SetFromCommandLine_ImExGus(SUNAdaptController C,
+                                                         const char* Cid,
+                                                         int argc, char* argv[])
+{
+  SUNFunctionBegin(C->sunctx);
+
+  int i;
+  SUNErrCode retval;
+  sunbooleantype write_parameters = SUNFALSE;
+  for (i = 1; i < argc; i++)
+  {
+    /* if Cid is supplied, skip command-line arguments that do not begin with Cid;
+       else, skip command-line arguments that do not begin with "sunadaptcontroller." */
+    size_t offset;
+    if (strlen(Cid) > 0)
+    {
+      if (strncmp(argv[i], Cid, strlen(Cid)) != 0) { continue; }
+      offset = strlen(Cid) + 1;
+    }
+    else
+    {
+      static const char* prefix = "sunadaptcontroller_imexgus.";
+      if (strncmp(argv[i], prefix, strlen(prefix)) != 0) { continue; }
+      offset = strlen(prefix);
+    }
+
+    /* control over SetParams function */
+    if (strcmp(argv[i] + offset, "params") == 0)
+    {
+      i += 1;
+      sunrealtype rarg1 = atof(argv[i]);
+      i += 1;
+      sunrealtype rarg2 = atof(argv[i]);
+      i += 1;
+      sunrealtype rarg3 = atof(argv[i]);
+      i += 1;
+      sunrealtype rarg4 = atof(argv[i]);
+      retval = SUNAdaptController_SetParams_ImExGus(C, rarg1, rarg2, rarg3,
+                                                    rarg4);
+      if (retval != SUN_SUCCESS) { return retval; }
+      continue;
+    }
+
+    /* control over SetDefaults function */
+    if (strcmp(argv[i] + offset, "defaults") == 0)
+    {
+      retval = SUNAdaptController_SetDefaults_ImExGus(C);
+      if (retval != SUN_SUCCESS) { return retval; }
+      continue;
+    }
+
+    /* control over SetErrorBias function */
+    if (strcmp(argv[i] + offset, "error_bias") == 0)
+    {
+      i += 1;
+      sunrealtype rarg = atof(argv[i]);
+      retval           = SUNAdaptController_SetErrorBias_ImExGus(C, rarg);
+      if (retval != SUN_SUCCESS) { return retval; }
+      continue;
+    }
+
+    /* check whether it was requested that all parameters be printed to screen */
+    if (strcmp(argv[i] + offset, "write_parameters") == 0)
+    {
+      write_parameters = SUNTRUE;
+      continue;
+    }
+  }
+
+  /* Call SUNAdaptController_Write_ImExGus (if requested) now that all
+     command-line options have been set -- WARNING: this knows
+     nothing about MPI, so it could be redundantly written by all
+     processes if requested. */
+  if (write_parameters)
+  {
+    retval = SUNAdaptController_Write_ImExGus(C, stdout);
```

Comment:
Same issue about moving this parsing up to superclass

Reply:
Same response as for `error_bias`

Reply:
Continuing the discussion in [this](https://github.com/LLNL/sundials/pull/662#discussion_r2072196366) thread